### PR TITLE
System Token improvements

### PR DESCRIPTION
**Pull request by Leandroqo**
- Changed auth's api to use hash instead crypto
- Update auth.conf to use regexMatch and keyMatch3 in path
- Added exclusive service to revoke token

### CODE REVIEW

File: gate/internal/repository/system_token_repository.go [hunk 1]
```diff
@@ -39,24 +37,16 @@ type SystemTokenRepository interface {
 	FindById(id uuid.UUID) (domain.SystemToken, error)
 	FindByToken(token string) (domain.SystemToken, error)
 	Update(systemToken domain.SystemToken) error
+	UpdateRevoke(systemToken domain.SystemToken) error
```
Comment (monicaribeiro) [inline gate/internal/repository/system_token_repository.go:40]: I would suggest changing the class name to 'UpdateRevokedStatus' or 'RevokeSystemToken'.

File: gate/internal/repository/system_token_repository.go [hunk 2]
```diff
@@ -162,6 +155,19 @@ func (systemTokenRepository systemTokenRepository) Update(systemToken domain.Sys
...
 }
 
+func (systemTokenRepository systemTokenRepository) UpdateRevoke(systemToken domain.SystemToken) error {
+	res := systemTokenRepository.db.
+		Table("system_tokens").
+		Where("id = ?", systemToken.ID).
+		UpdateColumns(models.SystemToken{Revoked: systemToken.Revoked, RevokedAt: systemToken.RevokedAt})
+
+	if res.Error != nil {
+		return handleSystemTokenError("Update system token failed", "SystemTokenRepository.UpdateRevoke.Updates", res.Error, logging.InternalError)
```
Comment (monicaribeiro) [inline gate/internal/repository/system_token_repository.go:165]:
```suggestion
		return handleSystemTokenError("Update system token failed", "SystemTokenRepository.UpdateRevoke.Update", res.Error, logging.InternalError)
```

* if the suggestion be applied to the method name, we need to remember to update here.

File: gate/resources/migrations/20210405150000_encrypt_system_token_value.up.sql [hunk 3]
```diff
@@ -1 +1 @@
-ALTER TABLE system_tokens ADD token bytea;
\ No newline at end of file
+ALTER TABLE system_tokens ADD token VARCHAR(64);
```
Comment (monicaribeiro) [inline gate/resources/migrations/20210405150000_encrypt_system_token_value.up.sql:1]: In this case, shouldn't we add another file to run the migration?

**Review by monicaribeiro — CHANGES_REQUESTED** [review]

**Review by barbararochazup — CHANGES_REQUESTED** [review]
Add in all changed files in header

Copyright 2020, 2021 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA

File: gate/resources/migrations/20210620143200_hash_system_token.up.sql [hunk 4]
```diff
@@ -0,0 +1,11 @@
+ALTER TABLE system_tokens
```
Comment (monicaribeiro) [inline gate/resources/migrations/20210620143200_hash_system_token.up.sql:1]: Do we have to add the license header here? @barbararochazup

**Review by monicaribeiro — CHANGES_REQUESTED** [review]

**Review by monicaribeiro — DISMISSED** [review]

**Review by barbararochazup — DISMISSED** [review]

**Review by guilhermemartinsdeoliveirazup — DISMISSED** [review]
PR reviewed with Barbara Rossalli.

**Review by thallesfreitaszup — APPROVED** [review]

**Review by guilhermemartinsdeoliveirazup — APPROVED** [review]

**Pull request merged**